src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
+            - wavelength : np.ndarray
+                1-D array representing the wavelengths of the sky model.
+            - intensity : np.ndarray
+                Array representing the intensity of the sky model.
+            - variance : np.ndarray
+                Array representing the variance associated with the sky model.
+            - verbose : bool
+                If True, print messages during execution. Default is True.
+        """
         self.wavelength = kwargs.get('wavelength', None)
         self.intensity = kwargs.get('intensity', None)
         self.variance = kwargs.get('variance', None)
         self.verbose = kwargs.get('verbose', True)
 
     def substract(self, data, variance, axis=-1, verbose=False):
-        """Substracts the sky_model to all fibres in the rss
+        """
+        Subtracts the sky model from the given data.
 
         Parameters
         ----------
-        data: (np.ndarray)
-            Data array for which the sky will be substracted
-        variance: (np.ndarray)
-            Array of variance data to include errors on determining the sky.
-        axis: (np.ndarray)
-            Spectral direction of data
+        data : np.ndarray
+            Data array from which the sky will be subtracted.
+        variance : np.ndarray
+            Array of variance data to include errors in determining the sky.
+        axis : int, optional
+            Spectral direction of the data. Default is -1.
+
         Returns
         -------
-        data_subs: (np.ndarray)
-        var_subs: (np.ndarray)
+        data_subs : np.ndarray
+            Data array after the sky model has been subtracted.
+        var_subs : np.ndarray
+            Variance array after including the sky model variance.
         """
-        # TODO
         if data.ndim == 3 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[:, np.newaxis, np.newaxis]
             skymodel_var = self.intensity[:, np.newaxis, np.newaxis]
         elif data.ndim == 2 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[np.newaxis, :]
             skymodel_var = self.intensity[np.newaxis, :]
+        elif data.ndim == 2 and self.intensity.ndim == 2:  
+            skymodel_intensity = self.intensity       
+            skymodel_var = self.variance
         else:
-            self.vprint(f"Data dimensions ({data.shape}) cannot be reconciled with sky mode ({self.intensity.shape})")
+            self.vprint(
+                f"Data dimensions ({data.shape}) cannot be reconciled with "
+                + f"sky mode ({self.intensity.shape})")
         data_subs = data - skymodel_intensity
         var_subs = variance + skymodel_var
         return data_subs, var_subs
 
     def substract_pca():
-        # TODO: Implement PCA substraction method
+        """
+        Placeholder for PCA subtraction method.
+
+        This method is not yet implemented.
+        """
         pass
```

Comment:
Does this method belong here?

Reply:
I believe so. To me, the way the sky model is created is independent of how it is substracted from the data. Any model could make use of a PCA. Otherwise, where do you think it could be implemented? In any case, this could be removed or commented out as it is not my intention to implement this feature on the near future.